src/lib/mdns/minimal/QueryReplyFilter.h

```diff
@@ -59,9 +57,41 @@ class QueryReplyFilter : public ReplyFilter
         return *this;
     }
 
+    QueryReplyFilter & SetSendingAdditionalItems(bool additional)
+    {
+        mSendingAdditionalItems = additional;
+        return *this;
+    }
+
 private:
+    bool AcceptableQueryType(QType qType)
+    {
+        if (mSendingAdditionalItems && mQueryData.IsBootAdvertising())
+        {
+            return true;
+        }
+
+        return ((mQueryData.GetType() == QType::ANY) || (mQueryData.GetType() == qType));
+    }
+
+    bool AcceptableQueryClass(QClass qClass)
+    {
+        return ((mQueryData.GetClass() == QClass::ANY) || (mQueryData.GetClass() == qClass));
+    }
+
+    bool AcceptablePath(FullQName qname)
+    {
+        if (mIgnoreNameMatch || mQueryData.IsBootAdvertising())
```

Comment:
Why do we need IsBootAdvertising instead of just using IgnoreNameMatch ?

Reply:
It seemed to make the intent cleaner by having behaviour depend on query.

The boot-time broadcaster builds a fake query (so that the sender sends replies to such a query) however it does not have access to the underlying filtering used by the mDNS sender.

Alternative would be to propagate flags into the mdns sending and that seeemed to be a bit uglier.